src/app/zap-templates/zcl/data-model/chip/chip-ota.xml

```diff
@@ -107,23 +107,32 @@ limitations under the License.
-    </enum>         
+    </enum>
+    <struct name="ProviderLocation">
+        <cluster code="0x002a"/>
+        <item name="FabricIndex" type="fabric_idx"/>
+        <item name="ProviderNodeID" type="node_id"/>
+        <item name="Endpoint" type="endpoint_no"/>
+    </struct>
     <cluster>
         <name>OTA Software Update Requestor</name>
         <domain>CHIP</domain>
-        <description>TODO</description>
+        <description>Provides an interface for downloading and applying OTA software updates</description>
         <code>0x002a</code>
         <define>OTA_REQUESTOR_CLUSTER</define>
         <client tick="false" init="false">true</client>
         <server tick="false" init="false">true</server>
-        <attribute side="server" code="0x0001" define="DEFAULT_OTA_PROVIDER" type="OCTET_STRING" length="16" writable="true"  optional="false">default ota provider</attribute>
-        <attribute side="server" code="0x0002" define="UPDATE_POSSIBLE"      type="BOOLEAN"                  writable="false" optional="false">update possible</attribute>
-        <command source="client" code="0" name="AnnounceOtaProvider" optional="true" cli="chip ota announceotaprovider">
-            <description>Notify OTA Provider that an update was applied</description>
-            <arg name="providerLocation" type="NODE_ID"/>
-            <arg name="vendorId" type="VENDOR_ID"/>
+        <attribute side="server" code="0x0000" define="DEFAULT_OTA_PROVIDERS" type="ARRAY" entryType="ProviderLocation" length="16" writable="true" optional="false">DefaultOtaProviders</attribute>
+        <attribute side="server" code="0x0001" define="UPDATE_POSSIBLE" type="BOOLEAN" default="true" writable="false" optional="false">UpdatePossible</attribute>
+        <attribute side="server" code="0x0002" define="UPDATE_STATE" type="UpdateStateEnum" default="Unknown" writable="false" optional="false">UpdateState</attribute>
+        <attribute side="server" code="0x0003" define="UPDATE_STATE_PROGRESS" type="INT8U" min="0" max="100" writable="false" optional="false">UpdateStateProgress</attribute>
+        <command source="client" code="0x00" name="AnnounceOtaProvider" optional="true" cli="chip ota announceotaprovider">
+            <description>Announce the presence of an OTA Provider</description>
+            <arg name="providerNodeId" type="node_id"/>
+            <arg name="vendorId" type="vendor_id"/>
             <arg name="announcementReason" type="OTAAnnouncementReason"/>
             <arg name="metadataForNode" type="OCTET_STRING" length="512" optional="true"/>
+            <arg name="endpoint" type="endpoint_no"/>
         </command>
         <event side="server" code="0x00" name="StateTransition" priority="info" optional="false">
           <description>This event SHALL be generated when a change of the UpdateState attribute occurs due to an OTA Requestor moving through the states necessary to query for updates.</description>
```

Comment:
Some of these event fields should be isNullable="true" too (PreviousState, TargetSoftwareVersion, ProgressPercent, PlatformCode).  Followup probably fine for this.

That said, the ProductId field of VersionApplied claims to be not nullable in the spec but have null as default value, which is a spec bug.  Please raise it?

Reply:
Filed https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/4793 to track the ProductID field bug in the spec